src/Simplex/Messaging/Agent.hs

```diff
@@ -1681,14 +1681,20 @@ synchronizeRatchet' c connId pqSupport' force = withConnLock c connId "synchroni
     _ -> throwE $ CMD PROHIBITED "synchronizeRatchet: not duplex"
 
 ackQueueMessage :: AgentClient -> RcvQueue -> SMP.MsgId -> AM ()
-ackQueueMessage c rq@RcvQueue {userId, server} srvMsgId = do
-  atomically $ incSMPServerStat c userId server ackAttempts
+ackQueueMessage c rq@RcvQueue {userId, connId, server} srvMsgId = do
+  incStat ackAttempts
   tryAgentError (sendAck c rq srvMsgId) >>= \case
-    Right _ -> atomically $ incSMPServerStat c userId server ackMsgs
-    Left (SMP _ SMP.NO_MSG) -> atomically $ incSMPServerStat c userId server ackNoMsgErrs
+    Right _ -> do
+      whenM (liftIO $ hasGetLock c rq) $ do
```

Comment:
we probably should rename hasGetLock and get locks to something more meaningful, like ntfLock or something, to show it's related to processing ntfs and mutually exclusive with subscriptions.